main.cpp

```diff
@@ -56,46 +56,71 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
+    T(int v, const char* p);  //1
     //2
+    int value;
     //3
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+T::T(int v, const char* p): value(v), name(p) {}
+
+struct Comparator                               //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }        
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float x { 0 }, y{ 0 };
+    float computeDistance(float* xUpdated)      //12
     {
-        
+        if (xUpdated != nullptr)
+        {
+            std::cout << "U's x value: " << this->x << std::endl;
+            this->x = *xUpdated;
+            std::cout << "U's y updated value: " << this->x << std::endl;
+            while( std::abs(this->y - this->x) > 0.001f )
+            {
+                this->y += .1f;
+            }
+            std::cout << "U's computeDistance updated value: " << this->y << std::endl;
+            return this->y * this->x;
+        }
+        std::cout << "Not allocated pointer " << std::endl;
+        return 0;
     }
 };
 
-struct <#structname2#>
+struct StaticU
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float computeDistance(U* that, float* xUpdated )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if (xUpdated != nullptr)
```

Comment:
Is there another pointer here that you should also be checking if it is nullptr?
- [ ] revise